data/binding/mapbinding.go

```diff
@@ -236,27 +272,97 @@ func (b *mapBase) setItem(key string, d DataItem) {
 	b.trigger()
 }
 
-func bindUntyped(m interface{}) Untyped {
-	return &boundUntyped{val: m}
+type boundStruct struct {
+	mapBase
+
+	orig interface{}
 }
 
-type boundUntyped struct {
+func (b *boundStruct) Reload() (retErr error) {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+
+	v := reflect.ValueOf(b.orig).Elem()
+	t := v.Type()
+	for j := 0; j < v.NumField(); j++ {
+		f := v.Field(j)
+		if !f.CanSet() {
+			continue
+		}
+
+		key := t.Field(j).Name
+		old := (*b.val)[key]
+		if f.Interface() == old {
+			continue
+		}
+
+		var err error
+		switch f.Kind() {
+		case reflect.Bool:
+			err = b.items[key].(*reflectBool).Set(f.Bool())
+		case reflect.Float32, reflect.Float64:
+			err = b.items[key].(*reflectFloat).Set(f.Float())
+		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+			err = b.items[key].(*reflectInt).Set(int(f.Int()))
+		case reflect.String:
+			err = b.items[key].(*reflectString).Set(f.String())
+		}
+		if err != nil {
+			retErr = err
+		}
+		(*b.val)[key] = f.Interface()
+	}
+	return
+}
+
+func bindUntypedMapValue(m *map[string]interface{}, k string, external bool) Untyped {
+	if external {
+		ret := &boundExternalMapValue{old: (*m)[k]}
+		ret.val = m
+		ret.key = k
+		return ret
+	}
+
+	return &boundMapValue{val: m, key: k}
+}
+
+type boundMapValue struct {
 	base
 
-	val interface{}
+	val *map[string]interface{}
+	key string
 }
 
-func (b *boundUntyped) get() (interface{}, error) {
-	return b.val, nil
+func (b *boundMapValue) get() (interface{}, error) {
+	if v, ok := (*b.val)[b.key]; ok {
+		return v, nil
+	}
+
+	return nil, errKeyNotFound
 }
 
-func (b *boundUntyped) set(val interface{}) error {
-	b.val = val
+func (b *boundMapValue) set(val interface{}) error {
+	(*b.val)[b.key] = val
 
 	b.trigger()
 	return nil
 }
 
+type boundExternalMapValue struct {
+	boundMapValue
+
+	old interface{}
+}
+
+func (b *boundExternalMapValue) setIfChanged(val interface{}) error {
+	if val == b.old {
```

Comment:
Is this guaranteed to be safe for any possible `interface{}`? My understanding is that an `interface{}` could include pointers to things, in which case simply comparing the pointer values may not be desired. We may want to use a "deep equals" here - there are libraries that can do this.

Reply:
That is a good point. Do we want to support that sort of data though?
If it turns out to be useful we could also check for pointer types and do clever things - ot just call without trying to do expensive deep check...